### PR TITLE
[INLONG-3161][DataProxy] Fix not responding client when throwing exception with order message

### LINKED ISSUE #3161: [Bug][Dataproxy] When sending order messages, if an exception is thrown, no response message is returned to client
### What happened

When sending order messages, if an exception is thrown, no response message is returned to client

### What you expected to happen

When sending order messages, if an exception is thrown, response message need to be returned to client

### How to reproduce

send order message , and thows exception using a incorrect url of pulsar server 

### Environment

_No response_

### InLong version

master

### InLong Component

InLong DataProxy

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by baomingyu**
Fixed a bug if an exception is thrown, no response message to client while using order message

Fixes #3161

### CODE REVIEW

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/pulsar/PulsarClientService.java [hunk 1]
```diff
@@ -246,6 +245,9 @@ public boolean sendMessage(int poolIndex, String topic, Event event,
      * @param orderEvent orderEvent
      */
     private void sendResponse(OrderEvent orderEvent) {
+        if ("false".equals(orderEvent.getHeaders().get(AttributeConstants.MESSAGE_IS_ACK))) {
```
Comment (healchow) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/pulsar/PulsarClientService.java:248]: Is it possible that the `orderEvent` is null?

File: inlong-common/src/main/java/org/apache/inlong/common/reporpter/AbstractReporter.java [hunk 2]
```diff
@@ -101,14 +101,15 @@ public Response syncReportData(T data, String serverUrl) throws Exception {
...
             StringEntity stringEntity = new StringEntity(gson.toJson(data));
             stringEntity.setContentType(AGENT_HTTP_APPLICATION_JSON);
             httpPost.setEntity(stringEntity);
-            String returnStr = executeHttpPost(httpPost);
+            returnStr = executeHttpPost(httpPost);
             return parse(returnStr);
         } catch (Exception e) {
-            LOGGER.error("syncReportData has exception e = {}", e);
+            LOGGER.error("syncReportData has exception returnStr = {}, e:", returnStr, e);
             throw e;
```
Comment (luchunliang) [inline inlong-common/src/main/java/org/apache/inlong/common/reporpter/AbstractReporter.java:113]: It is better to return a Response with error code.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/pulsar/PulsarClientService.java [hunk 3]
```diff
@@ -246,6 +245,9 @@ public boolean sendMessage(int poolIndex, String topic, Event event,
...
+            return;
```
Comment (luchunliang) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/pulsar/PulsarClientService.java:249]: How about "orderEvent.getHeaders().get(AttributeConstants.MESSAGE_IS_ACK))==null"?
It need to return directly also.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/ServerMessageHandler.java [hunk 4]
```diff
@@ -442,6 +442,10 @@ private void formatMessagesAndSend(ChannelHandlerContext ctx, Map<String, String
                     headers.put(AttributeConstants.DATA_TIME, String.valueOf(System.currentTimeMillis()));
                 }
 
+                if ("false".equals(commonAttrMap.get(AttributeConstants.MESSAGE_IS_ACK))) {
+                    headers.put(AttributeConstants.MESSAGE_IS_ACK, "false");
+                }
```
Comment (luchunliang) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/ServerMessageHandler.java:447]: How about "orderEvent.getHeaders().get(AttributeConstants.MESSAGE_IS_ACK))==null" and other value?

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/ServerMessageHandler.java [hunk 5]
```diff
@@ -525,7 +529,8 @@ private void responsePackage(ChannelHandlerContext ctx, Map<String, String> comm
             Channel remoteChannel,
             SocketAddress remoteSocketAddress,
             MsgType msgType) throws Exception {
-        if (!commonAttrMap.containsKey("isAck") || "true".equals(commonAttrMap.get("isAck"))) {
+        if (!commonAttrMap.containsKey(AttributeConstants.MESSAGE_IS_ACK)
+                || "true".equals(commonAttrMap.get(AttributeConstants.MESSAGE_IS_ACK))) {
```
Comment (luchunliang) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/ServerMessageHandler.java:533]: It is better. And 
String isAck = commonAttrMap.getOrDefault(AttributeConstants.MESSAGE_IS_ACK, Boolean.TRUE.toString());
if (Boolean.TRUE.toString().equals(isAck)) {

**Review by luchunliang — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/pulsar/PulsarClientService.java (hunk 1 above)
Comment (baomingyu) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/pulsar/PulsarClientService.java:248]: it never be null

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/pulsar/PulsarClientService.java (hunk 3 above)
Comment (baomingyu) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/pulsar/PulsarClientService.java:249]: just false need not to response message , both null ant true all need response.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/ServerMessageHandler.java (hunk 4 above)
Comment (baomingyu) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/ServerMessageHandler.java:447]: just false need to be handle.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/ServerMessageHandler.java (hunk 5 above)
Comment (baomingyu) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/ServerMessageHandler.java:533]: done

File: inlong-common/src/main/java/org/apache/inlong/common/reporpter/AbstractReporter.java (hunk 2 above)
Comment (baomingyu) [reply inlong-common/src/main/java/org/apache/inlong/common/reporpter/AbstractReporter.java:113]: Response protocol do not support error code felid now. Error code need be add later.

**Review by aloyszhang — APPROVED** [review]

**Pull request merged**